analyses/cell-type-ewings/scripts/select-cell-types.R

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env Rscript
+
+# This script is used to generate a table of reference cells for a given ScPCA libray that are annotated by 
+# SingleR and/or CellAssign to have a specific label
+# The output is a table that contains columns for barcodes, reference cell class, 
+# and then columns for each of the original cell type annotations for that cell 
+# The `reference_cell_class` column will contain either tumor or normal 
+
+project_root <- here::here()
+renv::load(project_root)
+
+suppressPackageStartupMessages({
+  library(optparse)
+  library(SingleCellExperiment)
+})
+
+option_list <- list(
+  make_option(
+    opt_str = c("--sce_file"),
+    type = "character",
+    help = "Path to RDS file containing a processed SingleCellExperiment object from scpca-nf"
+  ),
+  make_option(
+    opt_str = c("--normal_cells"),
+    type = "character",
+    default = NULL,
+    help = "Comma separated list of cell types to pull from annotations present and save as normal cells."
+  ),
+  make_option(
+    opt_str = c("--tumor_cells"),
+    type = "character",
+    default = NULL,
+    help = "Comma separated list of cell types to pull from annotations present and save as tumor cells."
+  ),
+  make_option(
+    opt_str = c("--output_filename"),
+    type = "character",
+    help = "Full path to file to store saved table of cell barcodes."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure path to sce file exists
+stopifnot("sce_file does not exist" = file.exists(opt$sce_file))
+
+# check that either normal or tumor cell types are provided 
+stopifnot("Either normal_cells or tumor_cells must be provided" = 
+            !is.null(opt$normal_cells) || !is.null(opt$tumor_cells))
```

Comment:
All checks can be done in a single call to `stopifnot()`
```suggestion
stopifnot(
  # make sure path to sce file exists
  "sce_file does not exist" = file.exists(opt$sce_file)),

  # check that either normal or tumor cell types are provided 
  "Either normal_cells or tumor_cells must be provided" = 
    !is.null(opt$normal_cells) || !is.null(opt$tumor_cells)
)       
```